### DISCUSSION TITLE
Question about absorption probabilities in a dataset with one terminal state

### DISCUSSION THREAD

**Discussion by natallah** [theislab/cellrank#867] [Q&A]
I have one terminal state in my population of cells, however I do not understand why when I call cellrank.pl.heatmap() I see a range of absorption probabilities in the results along the bar at the top of the figure and also in the annotation bar for each gene.  I am sure that I must be misunderstanding something obvious.

![macs_lineageDriversHeatmap](https://user-images.githubusercontent.com/7681553/165391731-fd943cc1-08f7-4ead-bc0c-c0e01b983236.png)

**Comment by Marius1311** [comment]
Hi @natallah, I think there's something odd going on here, check the scale on the color bar on the right, all values are really small. @michalk8, what exactly do we do when absorption probs are called for a single terminal state? I suppose they should all be set to exactly 1?

**Comment by natallah** [comment]
@Marius1311 ok I'm glad I asked then.  I guess I should also mention that the UMAPs of the driver genes are also confusing me (I thought that they were colored by probabilities also, and here too most of the probabilities are very small (and I am again confused as to why they are not all 1).


<img width="967" alt="macs_lineageDrivers" src="https://user-images.githubusercontent.com/7681553/165515217-692a7f00-ac2f-42cd-a879-da5f7fdd456c.png">

**Comment by michalk8** [comment]
> @michalk8, what exactly do we do when absorption probs are called for a single terminal state?

We use the stationary distribution.

**Reply by michalk8** [reply]
> I suppose they should all be set to exactly 1?

We could switch to that in `2.0` version, but in case of 1 terminal state, this would be meaningless, since we'd be correlating the expression with a constant values of 1.

>  I guess I should also mention that the UMAPs of the driver genes are also confusing me (I thought that they were colored by probabilities also, and here too most of the probabilities are very small

The UMAP is color by the  gene expression, sorted by their `q-value` (I think).

**Reply by natallah** [reply]
@michalk8 thank you - the UMAPs colored by expression makes sense.  So is the identification of driver genes essentially meaningless in a case such as this where the cells all have the same absorption probability essentially meaningless?

**Reply by Marius1311** [reply]
No - here's a short explanation of what's going on: usually, to compute driver genes, we correlate expression values of all genes against absorption probabilities, as outlined above. However, when we only have a single terminal state, it doesn't make sense to do that because all abs. probs. would be one. Instead, we use the stationary distribution and correlate with that. In v2 of CellRank, we plan to make this more obvious by issuing a warning about this. So what you currently get is not meaningless, it just has another interpretation: you see genes that correlate highly with the stationary distribution on the MC. Feel free to read up on stationary distributions for MC, there's plantly online. Hope this helped!
